### PR TITLE
DS-2115-v2 new endpoint for email participant auth

### PR DESCRIPTION
**Pull request by jmuto2**
## Description
new endpoint for email participant auth

## Jira Issue Link
https://jira.alldigitalrewards.com/browse/DS-2115

## Has the README / documentation been extended if necessary?
No

## Does this update require changes to public API documentation? 
Yes

### Are there created tests which fail without the change (if possible)?
No

### Are all tests passing?
Yes

### Have the changes been verified to comply with the security policy requirements?
Yes

### CODE REVIEW

File: src/Participant/EmailParticipantAuthRequest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * @throws Exception
+     */
+    public function __construct(string $programId, string $email, string $password)
+    {
+        $this->programId = $programId;
+        $this->email = trim($email);
+        $this->password = trim($password);
+        if (empty($this->email) === true) {
+            throw new Exception('Email must be valid.');
```
Comment (jhoughtelin) [inline src/Participant/EmailParticipantAuthRequest.php:35]: email must not be empty.

Comment (jhoughtelin) [reply src/Participant/EmailParticipantAuthRequest.php:35]: would be cool if it validated it as a real email tho'.

File: src/Participant/EmailParticipantAuthRequest.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    public function getResponseObject(): AbstractEntity
+    {
+        return new ParticipantRetrieveResponse();
+    }
+
+    public function jsonSerialize(): array
+    {
+        return [
+            'email_address' => $this->email,
```
Comment (jhoughtelin) [inline src/Participant/EmailParticipantAuthRequest.php:55]: this entity has a program id as well.

File: src/Participant/EmailParticipantAuthRequest.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private $password;
+
+    protected $httpMethod = 'POST';
+
+    /**
+     * @throws Exception
+     */
+    public function __construct(string $programId, string $email, string $password)
+    {
+        $this->programId = $programId;
```
Comment (jhoughtelin) [inline src/Participant/EmailParticipantAuthRequest.php:31]: Seems to be missing the empty check for programId, if you want to be consistent.

**Review by jhoughtelin — APPROVED** [review]

File: src/Participant/EmailParticipantAuthRequest.php (hunk 2 above)
Comment (jmuto2) [reply src/Participant/EmailParticipantAuthRequest.php:55]: not following sorry

**Pull request merged**